### ISSUE TITLE
ShellEditor issue

### ISSUE THREAD

**Issue by fred4ets** [enthought/traitsui#1625]
Hello,

I have the error message below when I run the following CME:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/traitsui-7.2.0.dev485-py3.7.egg/traitsui/qt4/ui_live.py", line 252, in _on_finished
    self.close(accept)
  File "/usr/local/lib/python3.7/dist-packages/traitsui-7.2.0.dev485-py3.7.egg/traitsui/qt4/ui_live.py", line 240, in close
    super().close(rc)
  File "/usr/local/lib/python3.7/dist-packages/traitsui-7.2.0.dev485-py3.7.egg/traitsui/qt4/ui_base.py", line 291, in close
    self.ui.dispose(rc)
  File "/usr/local/lib/python3.7/dist-packages/traitsui-7.2.0.dev485-py3.7.egg/traitsui/ui.py", line 256, in dispose
    self.finish()
  File "/usr/local/lib/python3.7/dist-packages/traitsui-7.2.0.dev485-py3.7.egg/traitsui/ui.py", line 278, in finish
    self.reset(destroy=False)
  File "/usr/local/lib/python3.7/dist-packages/traitsui-7.2.0.dev485-py3.7.egg/traitsui/ui.py", line 315, in reset
    editor.dispose()
  File "/usr/local/lib/python3.7/dist-packages/traitsui-7.2.0.dev485-py3.7.egg/traitsui/editors/shell_editor.py", line 136, in dispose
    self.update_object, "command_executed", remove=True, dispatch="ui"
  File "/usr/local/lib/python3.7/dist-packages/traits-6.3.0.dev1692-py3.7-linux-x86_64.egg/traits/has_traits.py", line 2330, in observe
    dispatcher=_ObserverDispatchers[dispatch],
  File "/usr/local/lib/python3.7/dist-packages/traits-6.3.0.dev1692-py3.7-linux-x86_64.egg/traits/observation/observe.py", line 59, in observe
    remove=remove,
  File "/usr/local/lib/python3.7/dist-packages/traits-6.3.0.dev1692-py3.7-linux-x86_64.egg/traits/observation/_observe.py", line 54, in add_or_remove_notifiers
    callable_()
  File "/usr/local/lib/python3.7/dist-packages/traits-6.3.0.dev1692-py3.7-linux-x86_64.egg/traits/observation/_observe.py", line 94, in __call__
    step()
  File "/usr/local/lib/python3.7/dist-packages/traits-6.3.0.dev1692-py3.7-linux-x86_64.egg/traits/observation/_observe.py", line 118, in _add_or_remove_extra_graphs
    remove=self.remove,
  File "/usr/local/lib/python3.7/dist-packages/traits-6.3.0.dev1692-py3.7-linux-x86_64.egg/traits/observation/_observe.py", line 54, in add_or_remove_notifiers
    callable_()
  File "/usr/local/lib/python3.7/dist-packages/traits-6.3.0.dev1692-py3.7-linux-x86_64.egg/traits/observation/_observe.py", line 94, in __call__
    step()
  File "/usr/local/lib/python3.7/dist-packages/traits-6.3.0.dev1692-py3.7-linux-x86_64.egg/traits/observation/_observe.py", line 150, in _add_or_remove_maintainers
    change_notifier.remove_from(observable)
  File "/usr/local/lib/python3.7/dist-packages/traits-6.3.0.dev1692-py3.7-linux-x86_64.egg/traits/observation/_observer_change_notifier.py", line 123, in remove_from
    raise NotifierNotFound("Notifier not found.")
traits.observation._exceptions.NotifierNotFound: Notifier not found.
```

Here's the CME:

```
#! /usr/bin/env python3

from traits.api import Any, Button, Dict, Instance, Int, Float, HasTraits
from traitsui.api import (Group, HGroup, Item, Tabbed, VGroup, UItem, View)
from traitsui.api import ShellEditor

class PythonShell(HasTraits):

    namespace = Dict

    def default_traits_view(self):
        view = View(UItem('namespace',
                          editor=ShellEditor(share=True)),
                    title='Python Shell Editor',
                    resizable=True,
                    scrollable=True,
                    buttons=['OK'])
        view.close_result = False
        return (view)

class Foo(HasTraits):

    shell_button = Button('run shell')
    python_shell = Instance(PythonShell)

    view = View(UItem('shell_button'),
                buttons=['Undo', 'OK', 'Cancel', 'Help'])


    # Traits default initialization ###

    def _python_shell_default(self):
        namespace = {}
        return (PythonShell(namespace=namespace))


    # Trait event handlers ###

    def _shell_button_fired(self):
        self.python_shell.edit_traits()


if (__name__ == '__main__'):
    foo = Foo()
    foo.configure_traits()
```

Thanks in advance for any help.
  
  
 Debian 10 x86_64, python 3.7.3
 
 I have this issue with Traits UI 7.2.0.dev485 but not with previous releases 7.2.0.dev447.

**Comment by rahulporuri** [comment]
thanks for the reporting the issue @fred4ets . we already know about this issue - see #1612 and #1620 . this is being fixed in #1623 . im closing this issue as a duplicate.

**Comment by fred4ets** [comment]
Great!

Thanks a lot.

Regards
